src/scene/materials/custom-material.js

```diff
@@ -0,0 +1,105 @@
+import { ShaderProcessorOptions } from '../../platform/graphics/shader-processor-options.js';
+import { LIGHTTYPE_DIRECTIONAL, LIGHTTYPE_OMNI, LIGHTTYPE_SPOT, MASK_AFFECT_DYNAMIC, SHADERDEF_INSTANCING, SHADERDEF_MORPH_NORMAL, SHADERDEF_MORPH_POSITION, SHADERDEF_MORPH_TEXTURE_BASED, SHADERDEF_SCREENSPACE, SHADERDEF_SKIN } from "../constants";
+import { getProgramLibrary } from "../shader-lib/get-program-library";
+import { LitOptions } from "./lit-options";
+import { collectLights } from "./lit-material-common.js";
+import { Material } from './material.js';
+import { custom } from '../shader-lib/programs/custom.js';
+
+/**
+ * A custom material provides a custom front-end part of a lit material
```

Comment:
adjust the description here to be more in line with `shaderChunk` property - so no `front-end part` for example.